### ISSUE TITLE
Can't compile bootloader on MSYS2 using mingw-w64 gcc and can't install PyInstaller

### ISSUE THREAD

**Issue by iheartcoffee** [pyinstaller/pyinstaller#4573]
When I try to install PyInstaller on MSYS2 I get the following error:
```
$pip install https://github.com/pyinstaller/pyinstaller/archive/develop.zip
Collecting https://github.com/pyinstaller/pyinstaller/archive/develop.zip
  Downloading https://github.com/pyinstaller/pyinstaller/archive/develop.zip
     - 23.2MB 417kB/s
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3.exe /usr/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /tmp/tmptlsrb1qh
       cwd: /tmp/pip-req-build-llhjfy8k
  Complete output (1 lines):
  Your platform is not yet supported. Please define constant PYDYLIB_NAMES for your platform.
  ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/python3.exe /usr/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /tmp/tmptlsrb1qh Check the logs for full command output.
```

So I thought maybe I need to compile the bootloader. But I get the following error:
```
$ python ./waf --gcc all
'all' finished successfully (0.001s)
'distclean' finished successfully (0.012s)
Setting top to                           : C:/msys64/home/jador/pyinstaller/bootloader
Setting out to                           : C:/msys64/home/jador/pyinstaller/bootloader/build
Python Version                           : 3.8.0 (default, Oct 25 2019, 06:23:40)  [GCC 9.2.0 64 bit (AMD64)]
Checking for program 'gcc, cc'           : C:\msys64\mingw64\bin/gcc.exe
Checking for program 'ar'                : C:\msys64\mingw64\bin/ar.exe
Checking size of pointer                 : 8
Platform                                 : Windows-64bit detected based on compiler
Checking for compiler flags -m64         : yes
Checking for program 'C:\\msys64\\mingw64\\bin/windres' : C:\msys64\mingw64\bin/windres.exe
Checking for library user32                             : yes
Checking for library comctl32                           : yes
Checking for library kernel32                           : yes
Checking for library ws2_32                             : yes
Checking for function unsetenv                          : no
Checking for function mkdtemp                           : no
Checking for function strndup                           : no
Checking for function strnlen                           : yes
Checking for program 'C:\\msys64\\mingw64\\bin/strip'   : C:\msys64\mingw64\bin/strip.exe
Checking for program 'strip'                            : C:\msys64\mingw64\bin/strip.exe
'configure' finished successfully (1.728s)
'make_all' finished successfully (0.020s)
Waf: Entering directory `C:/msys64/home/jador/pyinstaller/bootloader/build/debug'
[ 1/23] Compiling zlib/unzip.c
[ 2/23] Compiling zlib/inffast.c
[ 3/23] Compiling zlib/adler32.c
[ 4/23] Compiling zlib/ioapi.c
[ 5/23] Compiling zlib/inftrees.c
[ 6/23] Compiling zlib/crc32.c
[ 7/23] Compiling zlib/zutil.c
[ 8/23] Compiling zlib/inflate.c
[ 9/23] Compiling src/pyi_python.c
[10/23] Compiling src/pyi_archive.c
[11/23] Compiling src/pyi_launch.c
[12/23] Compiling src/pyi_main.c
[13/23] Compiling src/pyi_win32_utils.c
[14/23] Compiling src/pyi_global.c
[15/23] Compiling src/pyi_pythonlib.c
[16/23] Compiling src/pyi_utils.c
[17/23] Linking build/debug/libstatic_zlib.a
[18/23] Compiling windows/run.rc
C:\msys64\mingw64\bin/windres.exe: windows/run.rc.o: No such file or directory

../../src/pyi_pythonlib.c: In function 'pyi_pylib_start_python':
../../src/pyi_pythonlib.c:469:9: error: 'strncat' specified bound 1 equals source length [-Werror=stringop-overflow=]
  469 |         strncat(pypath, PYI_SEPSTR, strlen(PYI_SEPSTR));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:470:9: error: 'strncat' specified bound 16 equals source length [-Werror=stringop-overflow=]
  470 |         strncat(pypath, "base_library.zip", strlen("base_library.zip"));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:471:9: error: 'strncat' specified bound 1 equals source length [-Werror=stringop-overflow=]
  471 |         strncat(pypath, PYI_PATHSEPSTR, strlen(PYI_PATHSEPSTR));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:468:9: error: 'strncpy' specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
  468 |         strncpy(pypath, status->mainpath, strlen(status->mainpath));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:472:9: error: 'strncat' specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
  472 |         strncat(pypath, status->mainpath, strlen(status->mainpath));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:464:9: error: 'strncpy' specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
  464 |         strncpy(pypath, status->mainpath, strlen(status->mainpath));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1.exe: all warnings being treated as errors

Waf: Leaving directory `C:/msys64/home/jador/pyinstaller/bootloader/build/debug'
Build failed
 -> task in 'run_d' failed with exit status 1 (run with -v to display more information)
 -> task in 'run_d' failed with exit status 1 (run with -v to display more information)

```

I tried it without the gcc option
```
$ python ./waf all
'all' finished successfully (0.001s)
'distclean' finished successfully (0.022s)
Setting top to                           : C:/msys64/home/jador/pyinstaller/bootloader
Setting out to                           : C:/msys64/home/jador/pyinstaller/bootloader/build
Python Version                           : 3.8.0 (default, Oct 25 2019, 06:23:40)  [GCC 9.2.0 64 bit (AMD64)]
Checking for 'msvc' (C compiler)         : not found
Checking for 'gcc' (C compiler)          : C:\msys64\mingw64\bin/gcc.exe
Checking size of pointer                 : 8
Platform                                 : Windows-64bit detected based on compiler
Checking for compiler flags -m64         : yes
Checking for program 'C:\\msys64\\mingw64\\bin/windres' : C:\msys64\mingw64\bin/windres.exe
Checking for library user32                             : yes
Checking for library comctl32                           : yes
Checking for library kernel32                           : yes
Checking for library ws2_32                             : yes
Checking for function unsetenv                          : no
Checking for function mkdtemp                           : no
Checking for function strndup                           : no
Checking for function strnlen                           : yes
Checking for program 'C:\\msys64\\mingw64\\bin/strip'   : C:\msys64\mingw64\bin/strip.exe
Checking for program 'strip'                            : C:\msys64\mingw64\bin/strip.exe
'configure' finished successfully (1.742s)
'make_all' finished successfully (0.030s)
Waf: Entering directory `C:/msys64/home/jador/pyinstaller/bootloader/build/debug'
[ 1/23] Compiling zlib/unzip.c
[ 2/23] Compiling zlib/inffast.c
[ 3/23] Compiling zlib/adler32.c
[ 4/23] Compiling zlib/ioapi.c
[ 5/23] Compiling zlib/inftrees.c
[ 6/23] Compiling zlib/crc32.c
[ 7/23] Compiling zlib/zutil.c
[ 8/23] Compiling zlib/inflate.c
[ 9/23] Compiling src/pyi_python.c
[10/23] Compiling src/pyi_archive.c
[11/23] Compiling src/pyi_launch.c
[12/23] Compiling src/pyi_main.c
[13/23] Compiling src/pyi_win32_utils.c
[14/23] Compiling src/pyi_global.c
[15/23] Compiling src/pyi_pythonlib.c
[16/23] Compiling src/pyi_utils.c
[17/23] Linking build/debug/libstatic_zlib.a
[18/23] Compiling windows/run.rc
[19/23] Compiling src/main.c
[20/23] Compiling src/pyi_python27_compat.c
C:\msys64\mingw64\bin/windres.exe: windows/run.rc.o: No such file or directory

../../src/pyi_pythonlib.c: In function 'pyi_pylib_start_python':
../../src/pyi_pythonlib.c:469:9: error: 'strncat' specified bound 1 equals source length [-Werror=stringop-overflow=]
  469 |         strncat(pypath, PYI_SEPSTR, strlen(PYI_SEPSTR));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:470:9: error: 'strncat' specified bound 16 equals source length [-Werror=stringop-overflow=]
  470 |         strncat(pypath, "base_library.zip", strlen("base_library.zip"));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:471:9: error: 'strncat' specified bound 1 equals source length [-Werror=stringop-overflow=]
  471 |         strncat(pypath, PYI_PATHSEPSTR, strlen(PYI_PATHSEPSTR));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:468:9: error: 'strncpy' specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
  468 |         strncpy(pypath, status->mainpath, strlen(status->mainpath));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:472:9: error: 'strncat' specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
  472 |         strncat(pypath, status->mainpath, strlen(status->mainpath));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:464:9: error: 'strncpy' specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
  464 |         strncpy(pypath, status->mainpath, strlen(status->mainpath));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1.exe: all warnings being treated as errors

Waf: Leaving directory `C:/msys64/home/jador/pyinstaller/bootloader/build/debug'
Build failed
 -> task in 'run_d' failed with exit status 1 (run with -v to display more information)
 -> task in 'run_d' failed with exit status 1 (run with -v to display more information)
```

I tried with distclean
```
$ python ./waf distclean all
'distclean' finished successfully (0.023s)
'all' finished successfully (0.000s)
'distclean' finished successfully (0.001s)
Setting top to                           : C:/msys64/home/jador/pyinstaller/bootloader
Setting out to                           : C:/msys64/home/jador/pyinstaller/bootloader/build
Python Version                           : 3.8.0 (default, Oct 25 2019, 06:23:40)  [GCC 9.2.0 64 bit (AMD64)]
Checking for 'msvc' (C compiler)         : not found
Checking for 'gcc' (C compiler)          : C:\msys64\mingw64\bin/gcc.exe
Checking size of pointer                 : 8
Platform                                 : Windows-64bit detected based on compiler
Checking for compiler flags -m64         : yes
Checking for program 'C:\\msys64\\mingw64\\bin/windres' : C:\msys64\mingw64\bin/windres.exe
Checking for library user32                             : yes
Checking for library comctl32                           : yes
Checking for library kernel32                           : yes
Checking for library ws2_32                             : yes
Checking for function unsetenv                          : no
Checking for function mkdtemp                           : no
Checking for function strndup                           : no
Checking for function strnlen                           : yes
Checking for program 'C:\\msys64\\mingw64\\bin/strip'   : C:\msys64\mingw64\bin/strip.exe
Checking for program 'strip'                            : C:\msys64\mingw64\bin/strip.exe
'configure' finished successfully (1.731s)
'make_all' finished successfully (0.008s)
Waf: Entering directory `C:/msys64/home/jador/pyinstaller/bootloader/build/debug'
[ 1/23] Compiling zlib/unzip.c
[ 2/23] Compiling zlib/inffast.c
[ 3/23] Compiling zlib/adler32.c
[ 4/23] Compiling zlib/ioapi.c
[ 5/23] Compiling zlib/inftrees.c
[ 6/23] Compiling zlib/crc32.c
[ 7/23] Compiling zlib/zutil.c
[ 8/23] Compiling zlib/inflate.c
[ 9/23] Compiling src/pyi_python.c
[10/23] Compiling src/pyi_archive.c
[11/23] Compiling src/pyi_launch.c
[12/23] Compiling src/pyi_main.c
[13/23] Compiling src/pyi_win32_utils.c
[14/23] Compiling src/pyi_global.c
[15/23] Compiling src/pyi_pythonlib.c
[16/23] Compiling src/pyi_utils.c
[17/23] Linking build/debug/libstatic_zlib.a
[18/23] Compiling windows/run.rc
[19/23] Compiling src/main.c
C:\msys64\mingw64\bin/windres.exe: windows/run.rc.o: No such file or directory

../../src/pyi_pythonlib.c: In function 'pyi_pylib_start_python':
../../src/pyi_pythonlib.c:469:9: error: 'strncat' specified bound 1 equals source length [-Werror=stringop-overflow=]
  469 |         strncat(pypath, PYI_SEPSTR, strlen(PYI_SEPSTR));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:470:9: error: 'strncat' specified bound 16 equals source length [-Werror=stringop-overflow=]
  470 |         strncat(pypath, "base_library.zip", strlen("base_library.zip"));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:471:9: error: 'strncat' specified bound 1 equals source length [-Werror=stringop-overflow=]
  471 |         strncat(pypath, PYI_PATHSEPSTR, strlen(PYI_PATHSEPSTR));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:468:9: error: 'strncpy' specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
  468 |         strncpy(pypath, status->mainpath, strlen(status->mainpath));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:472:9: error: 'strncat' specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
  472 |         strncat(pypath, status->mainpath, strlen(status->mainpath));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:464:9: error: 'strncpy' specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
  464 |         strncpy(pypath, status->mainpath, strlen(status->mainpath));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1.exe: all warnings being treated as errors

Waf: Leaving directory `C:/msys64/home/jador/pyinstaller/bootloader/build/debug'
Build failed
 -> task in 'run_d' failed with exit status 1 (run with -v to display more information)
 -> task in 'run_d' failed with exit status 1 (run with -v to display more information)
```

What should I do? I know you can use PyInstaller with MSYS2

**Comment by danyeaw** [comment]
@seanmadawala You shouldn't need to compile anything to get PyInstaller working with MSYS2. Can you give me more information about your setup? It doesn't look like you are using a virtualenv, I would highly recommend that. Do you have pip installed from mingw-w64-x86_64-python3-pip, and did you try to upgrade it using pip?

**Comment by danyeaw** [comment]
This also seems like a duplicate of #4570. Can you please close one of them?

**Comment by danyeaw** [comment]
For the bootloader build issues you are seeing:
1. The issues about strncpy and strncat are in a separate issue #4419. I have created PR #4585 to fix this.
2. Issue #4501 covered the issue about `C:\msys64\mingw64\bin/windres.exe: windows/run.rc.o: No such file or directory` error. I can reproduce this issue in Python 3.8 (which isn't yet support by PyInstaller), and in Python 2.7. I have created PR #4586 to resolve this.

@iheartcoffee I haven't seen a response from your to resolve your issues running PyInstaller on MSYS2. Can we close this and #4570 if you aren't going to take the steps necessary to setup a virtualenv?
